domino/domino.py

```diff
@@ -525,15 +525,41 @@ def collaborators_get(self):
         url = self._routes.collaborators_get()
         return self._get(url)
 
-    def collaborators_add(self, usernameOrEmail, message=""):
+    def get_user_id(self, username_or_email): 
+        url = self._routes.users_get()
+        response = self.request_manager.get(url)
+        users = response.json()
+        user_from_user_name = [x for x in users if ('userName' in x) and (x['userName'] == username_or_email)]
+        user_from_email = [x for x in users if ('email' in x) and (x['email'] == username_or_email)]
+        user_list = user_from_user_name + user_from_email
+        if len(user_list) == 0:
+            raise Exception(f"Could not find user: {username_or_email}")
+        user = user_list[0]
+        userId = user['id']
+        return userId
+
+    def collaborators_add(self, username_or_email, message=""):
         self.requires_at_least("1.53.0.0")
-        url = self._routes.collaborators_add()
-        data = {
-            'collaboratorUsernameOrEmail': usernameOrEmail,
-            'welcomeMessage': message
+ 
+        user_id = self.get_user_id(username_or_email)
```

Comment:
On the other hand, I think it would be reasonable to raise an exception in the calling method here if trying to add a non-existent user as a collaborator -- but rather than a bare base Exception, we should define a new exception class in [domino/exceptions.py](https://github.com/dominodatalab/python-domino/blob/master/domino/exceptions.py), something like:
```python
class UserNotFoundException(DominoException):
    """User not found Exception"""
    pass
````
...and then raise it here with the friendly message.
```suggestion
        user_id = self.get_user_id(username_or_email)
        if user_id is None:
            raise UserNotFoundException(f"Could not add collaborator matching {username_or_email}")
```

Reply:
fixed.